driver-extras/src/main/java/com/datastax/driver/extras/codecs/ArrayCodecs.java

```diff
@@ -0,0 +1,583 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.extras.codecs;
+
+import java.lang.reflect.Array;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.InvalidTypeException;
+
+import static com.datastax.driver.core.ParseUtils.skipSpaces;
+
+/**
+ * A collection of convenience {@link TypeCodec} instances useful for
+ * serializing between CQL lists and Java arrays.
+ * <p>
+ * The codecs in this class provide the following mappings:
+ *
+ * <table summary="Supported Mappings">
+ *     <tr>
+ *         <th>Codec</th>
+ *         <th>CQL type</th>
+ *         <th>Java type</th>
+ *     </tr>
+ *     <tr>
+ *         <td>{@link ObjectArrayCodec}</td>
+ *         <td>{@code list<?>}</td>
+ *         <td>{@code T[]}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@link IntArrayCodec}</td>
+ *         <td>{@code list<int>}</td>
+ *         <td>{@code int[]}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@link LongArrayCodec}</td>
+ *         <td>{@code list<long>}</td>
+ *         <td>{@code long[]}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@link FloatArrayCodec}</td>
+ *         <td>{@code list<float>}</td>
+ *         <td>{@code float[]}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@link DoubleArrayCodec}</td>
+ *         <td>{@code list<double>}</td>
+ *         <td>{@code double[]}</td>
+ *     </tr>
+ * </table>
+ *
+ * Codecs in this class can either be registered individually, or,
+ * or they may all
+ * be registered via {@link #withPrimitiveArrayCodecs(CodecRegistry)}.
+ *
+ */
+public abstract class ArrayCodecs {
+
+    /**
+     * Registers {@link TypeCodec} instances in this class
+     * that handle primitive arrays with the given {@link CodecRegistry}.
+     *
+     * @param registry registry to add codecs to.
+     * @return updated registry with codecs.
+     */
+    public static CodecRegistry withPrimitiveArrayCodecs(CodecRegistry registry) {
+        return registry.register(IntArrayCodec.instance, LongArrayCodec.instance, FloatArrayCodec.instance, DoubleArrayCodec.instance);
+    }
```

Comment:
I'm not a big fan of this, because the client-side syntax is inconsistent:
```java
registry.register(new MyCustomCodec());
ArrayCodecs.withPrimitiveArrayCodecs(registry);
```
I'd rather have a method that returns an iterable, that we can pass to register:
```java
registry.register(new MyCustomCodec());
registry.register(ArrayCodecs.all());
```